src/xnnpack/node-type-defs.h

```diff
@@ -17,6 +17,7 @@ XNN_ENUM_ITEM(xnn_node_type_concatenate2, "Concatenate2")
 XNN_ENUM_ITEM(xnn_node_type_concatenate3, "Concatenate3")
 XNN_ENUM_ITEM(xnn_node_type_concatenate4, "Concatenate4")
 XNN_ENUM_ITEM(xnn_node_type_concatenate5, "Concatenate5")
+XNN_ENUM_ITEM(xnn_node_type_concatenate_n, "ConcatenateN")
```

Comment:
Just call it `concatenate`. Also, remove the `xnn_node_type_concatenateX` if they are unused now.